Adds two new research problems:  Waku incentivization and building applications on top of waku (#93)
vacp2p/vac.dev: open-problems.md

### Privacy-Preserving Incentivizations of waku protocols

Waku protocols are designed in such a way that resource-limited devices can make up for their lack of resources by relying on the more powerful peers in the network.
This can be seen at all the layers of the Waku protocol stack.
One of which is the Waku relay protocol in which peers voluntarily join the routing layer of Waku and aid message propagation.
While contributing peers may have altruistic reasons, or be a stakeholder in some higher-level applications and would like to deliver high-quality service via reliable infrastructure.
However, this cannot be generalized to all peers hence peers need to be incentivized for their participation and contribution in the network regardless of being tied to any specific application, or necessarily having altruistic wills.

> One of the most challenging protocols of Waku for incentivization is the Waku Relay protocol.

Reviewer: ```suggestion
One of the most challenging Waku protocols for incentivization is the Waku Relay protocol.
```

This is because the supplied service is in the form of bandwidth, and due to the Gossip-based nature of the routing protocol it is hard to verify which peers are the legit routers and hence are eligible for any designated reward.
Also, there is no trivial requester-responder relationship in that protocol. 
The reason is that every published message is routed by every other peer in the network.
Thus we face a one-to-many relationship.
In this research problem, researchers and engineers are asked to design and develop an incentivization mechanism that would account for the said issues and also is game theoretically sound, in the sense that all the participants would be rationally incentivized to follow the protocol description. 

### Devising novel applications of the Waku protocol stack

Waku provides a suite of modular p2p protocols to enable the realization of web2 applications in the decentralized web3 world in the form of DApps. 
Waku protocol stack features all the necessary components for the DApp developers namely, the networking/routing layer, and multiple discovery methods.

> It also provides various request-reply protocols to allow resource provisioning for resource-limited devices. 

Reviewer: ```suggestion
It also provides various request-response protocols to allow resource provisioning for resource-limited devices. 
```

> For example, store protocol (to account for storage limitations), and filter and light push protocols (to aid bandwidth-constrained peers).

Reviewer: ```suggestion
For example, store protocol (to account for storage limitations), and filter and lightpush protocols (to aid bandwidth-constrained peers).
```
